libraries/chain/include/eosio/chain/transaction.hpp

```diff
@@ -237,18 +235,20 @@ namespace eosio { namespace chain {
       using segment_type = fc::static_variant<digest_type, std::vector<char>>;
 
       struct partial {
+         // TODO: will need indication of what was pruned so correct exception can be thrown in apply_context.get_context_free_data
```

Comment:
We know if a segment of the context-free data was pruned if the `segment_type` holds the `digest_type` alternative.

Reply:
Oh, right. Thanks.